src/util.cpp

```diff
@@ -934,21 +935,15 @@ int GetNumCores()
     return boost::thread::physical_concurrency();
 }
 
-int getTrailingZeroBitsInByte(unsigned char inputByte)
+int getLeadingZeroBitsInByte(unsigned char inputByte)
 {
-    // output: c will count inputByte's trailing zero bits,
-    // so if inputByte is 1101000 (base 2), then c will be 3
-    int c = CHAR_BIT;
+    // Behavior of __builtin_clz() is undefined for zero input, so we need to handle this case explicitly.
+    if (inputByte == 0)
+        return 8;
 
-    if (inputByte)
-    {
-        inputByte = (inputByte ^ (inputByte - 1)) >> 1;  // Set inputByte's trailing 0s to 1s and zero rest
-        for (c = 0; inputByte; c++)
-        {
-            inputByte >>= 1;
-        }
-    }
-    return c;
+    // __builtin_clz() returns the number of leading zeros for an unsigned int, so we need to perform a
+    // conversion to use it for an unsigned char.
+    return __builtin_clz(inputByte) % (sizeof(unsigned int) * 8 - CHAR_BIT);
```

Comment:
Seems that `__builtin_clz` function is compiler and platform dependent and works with `unsigned int` type. Is it safe to use it? With trailing zeros before, we had safer solution, as for me.

Reply:
Implemented a new function with commit [db1a66f2005b9348c04ee9b06c1aef8907622837](https://github.com/HorizenOfficial/zen/commit/db1a66f2005b9348c04ee9b06c1aef8907622837).

I moved `__builtin_clz()` to a unit test to check the validity of the new implemented function.